RNSReloaded.DoubleTime/Mod.cs

```diff
@@ -0,0 +1,141 @@
+        if (
+            this.rnsReloadedRef != null
+            && this.rnsReloadedRef.TryGetTarget(out var rnsReloaded)
+            && this.hooksRef != null
+            && this.hooksRef.TryGetTarget(out var hooks)
+        ) {
+            rnsReloaded.LimitOnlinePlay();
+
+            string[] toHook = {
+                //"scr_enemy_add_pattern", //this desync cause only host gets this when a treasuresphere breaks?!
+                "scrdt_encounter",
+                "bp_wolf_steeltooth1",
+                "bp_wolf_steeltooth1_enrage",
+                "bp_wolf_steeltooth1_pt2",
+                "bp_wolf_steeltooth1_pt2_s",
+                //"bp_wolf_steeltooth1_pt3",
+                //"bp_wolf_steeltooth1_pt3_s",
+                "bp_wolf_steeltooth1_pt4",
+                "bp_wolf_steeltooth1_pt4_s",
+                //"bp_wolf_steeltooth1_pt5",
+                //"bp_wolf_steeltooth1_pt5_s",
+                "bp_wolf_steeltooth1_pt6",
+                "bp_wolf_steeltooth1_pt6_s",
+                //"bp_wolf_steeltooth1_pt7",
+                //"bp_wolf_steeltooth1_pt7_s",
+                "bp_wolf_steeltooth1_pt8",
+                "bp_wolf_steeltooth1_pt8_s",
+                //"bp_wolf_steeltooth1_pt9",
+                //"bp_wolf_steeltooth1_pt9_s",
+                "bp_wolf_steeltooth1_s"
+            };
+
+
+            foreach (var hookStr in toHook) {
+                RValue* Detour(
+                    CInstance* self, CInstance* other, RValue* returnValue, int argc, RValue** argv
+                ) {
+                    return this.AddPatternDetour(hookStr, self, other, returnValue, argc, argv);
+                }
+
+                var script = rnsReloaded.GetScriptData(rnsReloaded.ScriptFindId(hookStr) - 100000);
+
+                var hook = hooks.CreateHook<ScriptDelegate>(Detour, script->Functions->Function)!;
+
+                hook.Activate();
+                hook.Enable();
+
+                ScriptHooks[hookStr] = hook;
+
+            }
+
+
+            // refactor soonTM
+            RValue* Detour2(
+            CInstance* self, CInstance* other, RValue* returnValue, int argc, RValue** argv
+            ) {
+                return this.AddPatternDetour2(self, other, returnValue, argc, argv);
+            }
+
+            var script2 = rnsReloaded.GetScriptData(rnsReloaded.ScriptFindId("scrbp_zoom") - 100000);
+
+            var hook2 = hooks.CreateHook<ScriptDelegate>(Detour2, script2->Functions->Function)!;
+
+            hook2.Activate();
+            hook2.Enable();
+
+            ScriptHooks["scrbp_zoom"] = hook2;
+
+        }
+    }
+
+    private RValue* AddPatternDetour(
+        string name, CInstance* self, CInstance* other, RValue* returnValue, int argc, RValue** argv
+    ) {
+        var hook = ScriptHooks[name];
+        if (this.rnsReloadedRef!.TryGetTarget(out var rnsReloaded)) {
+            rnsReloaded.ExecuteScript("scrbp_gamespeed", self, other, [new RValue(this.config.SpeedMultiplier)]);
```

Comment:
Hmm. I might eventually change this so it works with other mods.

Reply:
Okay. i changed it. it is 10000x simpler now.

After finishing nightcore, I realized that I could do it this way. But I didn't want to change the code since people were already playing the old version. But I guess it's a good idea to change it now.

I thought it'd be more troublesome than this, somehow I had the notion that I had to check the incoming values and set a base case so it doesn't recursively hook... but no.. it's just this simple huh.

Reply:
oh wait. now it doesn't work with nightcore correctly. since they're both hooking the same script